modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/messaging/MailSenderMessageListener.java

```diff
@@ -171,10 +119,75 @@ protected InternetAddress[] filterInternetAddresses(
 		return filteredInternetAddresses.toArray(new InternetAddress[0]);
 	}
 
+	private void _processMailMessage(MailMessage mailMessage) throws Exception {
+		InternetAddress from = _filterInternetAddress(mailMessage.getFrom());
+
+		if (from == null) {
+			if (_log.isWarnEnabled()) {
+				_log.warn("Skipping email because the sender is not specified");
+			}
+
+			return;
+		}
+
+		mailMessage.setFrom(from);
+
+		InternetAddress[] to = _filterInternetAddresses(mailMessage.getTo());
+
+		mailMessage.setTo(to);
+
+		InternetAddress[] cc = _filterInternetAddresses(mailMessage.getCC());
+
+		mailMessage.setCC(cc);
+
+		InternetAddress[] bcc = _filterInternetAddresses(mailMessage.getBCC());
+
+		InternetAddress[] auditTrail = InternetAddress.parse(
+			PropsValues.MAIL_AUDIT_TRAIL);
+
+		if (auditTrail.length > 0) {
+			if (ArrayUtil.isNotEmpty(bcc)) {
+				for (InternetAddress internetAddress : auditTrail) {
+					bcc = ArrayUtil.append(bcc, internetAddress);
+				}
+			}
+			else {
+				bcc = auditTrail;
+			}
+		}
+
+		mailMessage.setBCC(bcc);
+
+		InternetAddress[] bulkAddresses = _filterInternetAddresses(
+			mailMessage.getBulkAddresses());
+
+		mailMessage.setBulkAddresses(bulkAddresses);
+
+		InternetAddress[] replyTo = _filterInternetAddresses(
+			mailMessage.getReplyTo());
+
+		mailMessage.setReplyTo(replyTo);
+
+		if (ArrayUtil.isNotEmpty(to) || ArrayUtil.isNotEmpty(cc) ||
+			ArrayUtil.isNotEmpty(bcc) || ArrayUtil.isNotEmpty(bulkAddresses)) {
+
+			MailEngine.send(mailMessage);
+		}
+	}
+
+	private void _processMethodHandler(MethodHandler methodHandler)
+		throws Exception {
+
+		methodHandler.invoke(HookFactory.getInstance());
+	}
+
 	private static final Log _log = LogFactoryUtil.getLog(
-		MailMessageListener.class);
+		MailSenderMessageListener.class);
 
 	private static final Set<String> _mailSendBlacklist = new HashSet<>(
 		Arrays.asList(PropsValues.MAIL_SEND_BLACKLIST));
 
+	@Reference
+	private MailSender _mailSender;
```

Comment:
@julschong This reference is not used. Just to make sure, did you add it to ensure the listener starts after the `MailSenderImpl` OSGi service?

Reply:
will do. No, I'm pretty sure I did thing by mistake